src/jdk.httpserver/share/classes/sun/net/httpserver/ServerImpl.java

```diff
@@ -440,11 +486,11 @@ void reRegister (HttpConnection c) {
         }
 
         public void run() {
-            while (!finished) {
+             while (!isFinishing()) {
```

Comment:
The name of the method is not great - but since it's pre-existing I'm OK with that.

```suggestion
            // isFinishing() will be true when there are no active exchange after terminating
             while (!isFinishing()) {
```